app/dashboard/models.py

```diff
@@ -2400,6 +2430,136 @@ def is_staff(self):
+        """ Determines the longest streak, in workdays, of this user
+
+        Returns:
+            int: a number of weekdays
+
+        """
+
+        # setup
+        action_dates = self.actions.all().values_list('created_on', flat=True)
+        start_date = timezone.datetime(self.created_on.year, self.created_on.month, self.created_on.day)
+        end_date = timezone.datetime(timezone.now().year, timezone.now().month, timezone.now().day)
+
+        # loop setup
+        iterdate = start_date
+        max_streak = 0
+        this_streak = 0
+        while iterdate < end_date:
+            # housekeeping
+            last_iterdate = start_date
+            iterdate += timezone.timedelta(days=1)
+
+            is_weekday = iterdate.weekday() < 5
+            if not is_weekday:
+                continue
+
+            has_action_during_window = len([ele for ele in action_dates if ele > last_iterdate.replace(tzinfo=pytz.utc) and ele < iterdate.replace(tzinfo=pytz.utc)])
+            if has_action_during_window:
+                this_streak += 1
+                if this_streak > max_streak:
+                    max_streak = this_streak
+            else:
+                this_streak = 0
+
+        return max_streak
+
+    def calc_num_repeated_relationships(self):
+        """ the number of repeat relationships that this user has created
+
+        Returns:
+            int: a number of repeat relationships
+
+        """
+        bounties = self.bounties
+        completed_bounties = bounties.filter(idx_status__in=['done'])
+        relationships = []
+        for bounty in completed_bounties:
+            fulfiller_handles = bounty.fulfillers_handles
+            if bounty.is_funder(self.handle):
+                for handle in fulfiller_handles:
+                    relationships.append(handle)
+            if self.handle in fulfiller_handles:
+                relationships.append(bounty.bounty_owner_github_username)
+
+        rel_count = { key: 0 for key in relationships }
+        for rel in relationships:
+            rel_count[rel] += 1
+
+        return len([key for key, val in rel_count.items() if val > 1])
+
+
+    def calc_avg_hourly_rate(self):
+        """
+
+        Returns:
+            float: the average hourly rate for this user in dollars
+
+        """
+        rates = []
+        for bounty in self.bounties:
+            hr = bounty.hourly_rate
+            if hr:
+                rates.append(hr)
+        if len(rates) == 0:
+            return 0
+        return sum(rates)/len(rates)
+
+    def calc_success_rate(self):
+        """
+
+        Returns:
+            int; the success percentage for this users bounties as a positive integer.
+
+        """
+        bounties = self.bounties
+        completed_bounties = bounties.filter(idx_status__in=['done'])
+        eligible_bounties = bounties.filter(idx_status__in=['done', 'expired', 'cancelled'])
+
+        if eligible_bounties.count() == 0:
+            return -1
```

Comment:
hm why -1?

Reply:
its a way of saying "this user hasnt done any bounties; pls dont display on frontend". as opposed to 0 which is for when the user has made $0/hr for many bounties